global-configuration/src/main/resources/archetype-resources/src/test/java/SampleConfigurationTest.java

```diff
@@ -23,23 +23,46 @@ public class SampleConfigurationTest {
      * </ul>
      */
     @Test
-    public void uiAndStorage() throws Throwable {
-        sessions.then(r -> {
-            assertNull("not set initially", SampleConfiguration.get().getLabel());
-            HtmlForm config = r.createWebClient().goTo("configure").getFormByName("config");
+    void uiAndStorage(JenkinsRule jenkins) throws Throwable {
+        assertNull(SampleConfiguration.get().getLabel(), "not set initially");
+        try (JenkinsRule.WebClient client = jenkins.createWebClient()) {
+            HtmlForm config = client.goTo("configure").getFormByName("config");
             HtmlTextInput textbox = config.getInputByName("_.label");
             textbox.setText("hello");
-            r.submit(config);
-            assertEquals(
-                    "global config page let us edit it",
-                    "hello",
-                    SampleConfiguration.get().getLabel());
-        });
-        sessions.then(r -> {
-            assertEquals(
-                    "still there after restart of Jenkins",
-                    "hello",
-                    SampleConfiguration.get().getLabel());
-        });
+            jenkins.submit(config);
+            assertEquals("hello", SampleConfiguration.get().getLabel(), "global config page let us edit it");
+        }
+
+        jenkins = restartJenkins(jenkins);
+
+        assertEquals("hello", SampleConfiguration.get().getLabel(), "still there after restart of Jenkins");
+
+        FileUtils.deleteQuietly(jenkins.getInstance().getRootDir());
+    }
+
+    private static JenkinsRule restartJenkins(JenkinsRule currentJenkinsRule) throws Throwable {
```

Comment:
I do not feel comfortable yet recommending this as a general pattern in the archetype because it involves copying and pasting this code into every plugin that wants to use this feature with JUnit 5. The existing JUnit 4 functionality is generalized and easy-to-use, and I think its JUnit 5 equivalent should be equally generalized and easy-to-use without copying and pasting code like this into every Jenkins plugin.

Reply:
100% agree on that, should have converted to draft. I was just looking for a solution that works at all. There are quite some roadblocks ahead to make JUnit5 work nicely with the current `JenkinsRule` implementation.